app/src/lib/stores/helpers/ahead-behind-updater.ts

```diff
@@ -100,25 +100,22 @@ export class AheadBehindUpdater {
     this.clear()
 
     const from = currentBranch.tip.sha
+    const cache = this.comparisonCache
+    const newRefs = new Set<string>()
 
-    const filterBranchesNotInCache = (branches: ReadonlyArray<Branch>) => {
-      return branches
-        .map(b => b.tip.sha)
-        .filter(to => !this.comparisonCache.has(from, to))
+    const addBranch = (b: Branch | null) => {
+      if (b && b.isDesktopForkRemoteBranch && !cache.has(from, b.tip.sha)) {
+        newRefs.add(b.tip.sha)
+      }
     }
 
-    const otherBranches = [...recentBranches, ...allBranches]
-
-    const branches =
-      defaultBranch !== null ? [defaultBranch, ...otherBranches] : otherBranches
-
-    const newRefsToCompare = new Set<string>(filterBranchesNotInCache(branches))
+    addBranch(defaultBranch)
+    recentBranches.forEach(addBranch)
+    allBranches.forEach(addBranch)
```

Comment:
Assuming I understand the intent of this, and given I needed to look for more implementation details, it's probably worth commenting here that we want recent branches to be scheduled first (since they show up first in the UI), and it's not a problem scheduling them again later with `allBranches` because they'll be in cache already and no extra work will be done.

Does that sound good?

Reply:
> because they'll be in cache already and no extra work will be done.

And also because the `newRefs` is a set consisting of the tip SHA of each branch they won't actually be scheduled to be refreshed. I'm proposing we don't bother too much with this and instead focus on reviewing and merging #11160 which will remove the ahead-behind-updater entirely.

Reply:
Yep, I'm reviewing that one as we speak!